src/component/JoinList.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from "react";
+import { withFirestore } from "react-firestore";
+import { Redirect } from 'react-router-dom'
+
+const JoinList = ({ firestore }) => {
+  const [token, setToken] = useState("");
+  const [redirect, setRedirect] = useState(false)
+
+  const renderRedirect = () => {
+    if (redirect) {
+      return <Redirect to='/list' />
+    }
+  }
+
+  const checkTokenExists = token => {
+    firestore
+      .collection('lists')
+      .doc(token)
+      .onSnapshot(snapshot => {
+        if (snapshot.exists) {
```

Comment:
Curious about these two lines. I'm sure they make perfect sense, just trying to wrap my mind around it. Docs say onSnapshot is used like a document event listener. Is it so that if the list gets updated by another user the changes will show up without reloading the page?

Reply:
I might let @michellejae explain this more :) My understanding is just that if the list doesn't exist in the document, that snapshot isn't going to exist, so we know that it's not a valid token.

Reply:
@ClaudiaStar in theory, yes that is what is supposed to happen, ha. onSnapshot is supposed to be updating whenever the lists is updated in the DB. however, i'm not sure how that works with react state and hooks or if it's even effected. i can go ahead and change it to .get which is just a one time listener, as this component is purely just to check if the token exists. in doesn't actually care about the list on said token being updated as that is handled by the redirect back to the list component.